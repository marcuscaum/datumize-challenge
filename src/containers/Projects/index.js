import { connect } from 'react-redux';

import { getProjects, getRoles, getUsers } from '../../redux/selectors';
import {
  createActions,
  deleteActions,
  assignUserToProject,
  fetchProjects,
} from '../../redux/actions';

import ProjectsPage from '../../pages/Projects';

const mapStateToProps = state => ({
  projects: getProjects(state),
  roles: getRoles(state),
  users: getUsers(state),
});

const mapDispatchToProps = {
  createProject: createActions.project,
  deleteProject: deleteActions.project,
  assignUserToProject,
  fetchProjects,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ProjectsPage);
