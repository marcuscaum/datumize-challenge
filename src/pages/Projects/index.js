import React from 'react';

const Projects = () => (
  <React.Fragment>
    <header>
      <h1> Projects </h1>
    </header>
    <section>
      <ul>
        <li>ProjectItem 1</li>
        <li>ProjectItem 2</li>
      </ul>
    </section>
  </React.Fragment>
);

export default Projects;
