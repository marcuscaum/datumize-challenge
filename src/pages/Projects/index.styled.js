import styled from 'styled-components';

export const RolesList = styled.div`
  display: flex;
  margin: 15px 0;
  font-size: 12px;
  margin-bottom: 5px;
`;

export const RoleItem = styled.div`
  margin-right: 10px;
  background-color: #f3f3f3;
  padding: 5px 10px;
  border-radius: 5px;

  strong {
    margin-right: 5px;

    &::after {
      content: ':';
    }
  }
`;

export const RoleButton = styled.div`
  border: 0;
  padding: 0;
  text-decoration: underline;
  float: right;
  color: #004257;
  cursor: pointer;
`;
