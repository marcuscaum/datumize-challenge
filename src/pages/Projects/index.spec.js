import React from 'react';
import { shallow } from 'enzyme';

import Projects from './index';

const setup = props => shallow(<Projects {...props} />);

describe('[Page] Projects', () => {
  it('should render the page', () => {
    const wrapper = setup({
      fetchProjects: () => jest.fn(),
      fetchRoles: () => jest.fn(),
    });
    expect(wrapper).toMatchSnapshot();
  });
});
