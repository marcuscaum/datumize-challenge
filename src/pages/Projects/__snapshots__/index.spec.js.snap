// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] Projects should render the page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <lifecycle(withState(Component))
    fetchProjects={[Function]}
    fetchRoles={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "fetchProjects": [Function],
      "fetchRoles": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "fetchProjects": [Function],
        "fetchRoles": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
